PWGJE/FlavourJetTasks/AliAnalysisTaskHFJetIPQA.cxx

```diff
@@ -2892,303 +2839,184 @@ Bool_t AliAnalysisTaskHFJetIPQA::myquarksort(const SQuarks& i, const SQuarks& j)
     {
         return kTRUE;
     }
+    /*! \brief IsMCJetPartonFast
+     *
+     * Fast jet parton MC matcher
+     */
+        Int_t  AliAnalysisTaskHFJetIPQA::IsMCJetPartonFast(const AliEmcalJet *jet, Double_t radius,Bool_t &is_udg)
+        {
+            fJetCont.clear();
+            fPUdsgJet.clear();
+            fPSJet.clear();
+            fPCJet.clear();
+            fPBJet.clear();
+            daughtermother.clear();
+
+            double p_udsg_max=-999;
+            double p_s_max=-999;
+            AliVParticle *vp=0x0;
+            Int_t pdg=0;
+            Double_t p=0;
+            int kJetOrigin=-999;
+
+            //printf("XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX\n");
+            //printf("Starting loop\n");
+            if(!jet) return 0;
+            if(!(jet->GetNumberOfTracks()>fNoJetConstituents)){
+              //printf("Throwing away jets with only too few constituent!\n");
+              return 0;
+            }
 
-/*! \brief IsParton
- *
- * Return kTRUE if particle is parton
- */
-Bool_t AliAnalysisTaskHFJetIPQA::IsParton(int pdg){
-  return ((pdg==1)||(pdg==2)||(pdg==3)||(pdg==4)||(pdg==5)||(pdg==21));
-}
+            if(fDoFlavourMatching){
+              for(UInt_t i = 0; i < jet->GetNumberOfTracks(); i++) {//start trackloop jet
+                vp = static_cast<AliVParticle*>(jet->Track(i));
+                if (!vp){
+                  Printf("ERROR: AliVParticle associated to constituent not found\n");
```

Comment:
Error logging should be better done with `AliError()` or `AliErrorStream ()<<` as they can be steered with log levels, but not urgent.